src/app/certificado/components/certificado-form/certificado-form.component.ts

```diff
@@ -0,0 +1,195 @@
+import { Component, OnInit } from '@angular/core';
+import { Router, ActivatedRoute, Params } from '@angular/router';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+
+import { CertificadoService } from '../../../core/services/certificado.service';
+import { CertificateState, CertificateType } from '../../../core/models/certificado.model';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+@Component({
+  selector: 'app-certificado-form',
+  templateUrl: './certificado-form.component.html',
+  styleUrls: ['./certificado-form.component.scss'],
+})
+export class CertificadoFormComponent implements OnInit {
+  public form: FormGroup;
+  public form2: FormGroup;
+  public isCreate: boolean;
+  public tipo: boolean;
+  public nameFile: string;
+  public idFile: number;
+  private file: FormData;
+
+  public types2 = ['CA_NV', 'NV', 'CA_DEF', 'DEF'];
+
+  constructor(
+    private router: Router,
+    private builder: FormBuilder,
+    private certificadoService: CertificadoService,
+    private activateRoute: ActivatedRoute,
+    private _snackBar: MatSnackBar
+  ) {
+    this.buildForm();
+    this.buildFormFile();
+  }
+
+  ngOnInit(): void {
+    this.activateRoute.params.subscribe((params: Params) => {
+      const number = params.number;
+      if (number) {
+        this.isCreate = false;
+        this.certificadoService.findByID(number).subscribe((resp) => {
+          this.idFile = resp['number'];
+          if (resp['state'] === 'IDLE' || resp['state'] === 'ASSIGNED' || resp['state'] === 'GUARDED') {
+            this.tipo = false;
+          } else {
```

Comment:
This else is left over if you set the variable to true from the start, or you should change the validation